### PR TITLE
ICU-21900 Fix ICU4C sample build and execution

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21900
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/samples/datecal/cal.cpp [hunk 1]
```diff
@@ -17,6 +17,8 @@
 #include "unicode/gregocal.h"
 #include <stdio.h>
 
+using namespace icu;
```
Comment (jefgen) [inline icu4c/source/samples/datecal/cal.cpp:20]: Interesting that I didn't see any warnings when building the samples with MSVC...
regardless, thanks for finding and fixing this.

File: icu4c/source/samples/Makefile.in [hunk 2]
```diff
@@ -22,7 +22,8 @@ include @platform_make_fragment@
 ## Files to remove for 'make clean'
 CLEANFILES = *~
 
-SUBDIRS = date cal
+SUBDIRS = cal citer coll csdet date datecal ufortune uresb datefmt msgfmt numfmt props strsrch translit uciter8 ucnv udata ustring dtitvfmtsample dtptngsample plurfmtsample
+
 ALLSUBDIRS = break case csdet datefmt msgfmt numfmt props translit ucnv udata ufortune uresb ustring citer uciter8 ugrep
```
Comment (jefgen) [inline icu4c/source/samples/Makefile.in:27]: Should these be in `ALLSUBDIRS` ? (Not sure)

**Review by jefgen — COMMENTED** [review]
Hm, it looks like the build bots aren't happy:

```
make[1]: Making `all' in `citer'
/bin/bash: line 10: cd: citer: No such file or directory
make[1]: *** [Makefile:57: all-recursive] Error 1
make[1]: Leaving directory '/home/runner/work/icu/icu/build/samples'
```

File: icu4c/source/samples/datecal/cal.cpp (hunk 1 above)
Comment (sven-oly) [reply icu4c/source/samples/datecal/cal.cpp:20]: I've been fixing other compile errors and warnings, too. Different compilers!

File: icu4c/source/samples/Makefile.in (hunk 2 above)
Comment (sven-oly) [reply icu4c/source/samples/Makefile.in]: I see no reason for ALLSUBDIRS. I've removed this definition and use.

**Comment by sven-oly** [comment]
I think this is ready to submit. Build now works.

**Review by jefgen — APPROVED** [review]
Thanks for all this clean up and fixing things. :)

**Pull request merged**